Allow repetitions=0 in cirq.sample

viathor opened the pull request:
Fixes #2154 and #2196.

vtomole commented on cirq/study/trial_result.py:
```diff
@@ -108,7 +108,7 @@ def data(self) -> pd.DataFrame:
                     value.big_endian_bits_to_int(m_vals) for m_vals in val
                 ]
-            self._data = pd.DataFrame(converted_dict)
+            self._data = pd.DataFrame(converted_dict, dtype=np.int64)
```
Could there be any problems in the future from hard coding this to int64?

viathor replied:
See other comment.

kevinsung commented on cirq/study/trial_result.py:
```diff
@@ -108,7 +108,7 @@ def data(self) -> pd.DataFrame:
                     value.big_endian_bits_to_int(m_vals) for m_vals in val
                 ]
-            self._data = pd.DataFrame(converted_dict)
+            self._data = pd.DataFrame(converted_dict, dtype=np.int64)
```
Doesn't this limit us to 63 qubits?

viathor replied:
Pandas is actually smart about this. It'll use the suggested type when possible, but if you pass values that are too large, it'll upgrade to `dtype='object'`:

```
In [2]: pd.DataFrame({'a': []}, dtype=np.int64).to_numpy().dtype
Out[2]: dtype('int64')

In [3]: pd.DataFrame({'a': [1, 2]}, dtype=np.int64).to_numpy().dtype
Out[3]: dtype('int64')

In [4]: pd.DataFrame({'a': [1, 2**64+1]}, dtype=np.int64).to_numpy().dtype
Out[4]: dtype('O')
```

I've changed to `int` since it is less misleading.

viathor commented:
I think the Windows failures were actually caused by my change from dtype=np.int64 to dtype=int:

```
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  int32
E       [right]: int64
```

This suggests that specifying `int` allows pandas to use 32-integers which our tests don't accept.

I switched back to dtype=np.int64 and added a comment explaining that pandas will use dtype=object when np.int64 turns out insufficient.
